app/src/main/java/com/rage/clamber/Fragments/UserInfoFragment.java

```diff
@@ -39,19 +70,95 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
 
-        //TODO: Determine how to layout the recommended climbs with wall info
-//        recyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));
-//        ClimbsRecyclerViewAdapter adapter = new ClimbsRecyclerViewAdapter(CLIMB_NUMS);
-//        recyclerView.setAdapter(adapter);
-//        recyclerView.setItemAnimator(new DefaultItemAnimator());
+        mainUser = getArguments().getParcelable(HomePage.ARG_USER);
+
+        climbList = new ArrayList<>();
+
+        layoutId = R.id.home_page_frame_layout;
+        getCompletedClimbsForUser(mainUser.getUserName());
+
+        recyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));
+        adapter = new ClimbsRecyclerViewAdapter(climbList, mainUser, getActivity(), layoutId);
+        recyclerView.setAdapter(adapter);
 
         return rootView;
     }
 
 
-    @OnClick(R.id.user_activity_user_info_button)
-    public void onClickButtonClicked(Button button) {
-        NewUserDialogFragment fragment = new NewUserDialogFragment();
-        fragment.show(getActivity().getSupportFragmentManager(), "dialog");
+    /**
+     * Method to get all of the climbs that have been marked as completed by the user.
+     *
+     * @param username - username for the query.
+     */
+    public void getCompletedClimbsForUser(String username) {
+        ConnectivityManager connectivityManager = (ConnectivityManager) getActivity().getSystemService(Context.CONNECTIVITY_SERVICE);
+        NetworkInfo networkInfo = connectivityManager.getActiveNetworkInfo();
+        if (networkInfo != null && networkInfo.isConnected()) {
+            Call<List<Climb>> completedCall = ApiManager.getClamberService().getCompletedForUser(username);
+            completedCall.enqueue(new Callback<List<Climb>>() {
+                @Override
+                public void onResponse(Call<List<Climb>> call, Response<List<Climb>> response) {
+                    if (response.code() == 200) {
+                        List<Climb> climbs = response.body();
+                        climbList.addAll(climbs);
+                    } else {
+                        Log.d(TAG, "Non 200 response code returned - check server");
+                    }
+                    adapter.notifyDataSetChanged();
+                }
+
+                @Override
+                public void onFailure(Call<List<Climb>> call, Throwable t) {
+                    Log.d(TAG, "Failure when attempting to return projects", t);
+                }
+            });
+        } else {
+            Toast.makeText(getContext(), R.string.no_internet_connection, Toast.LENGTH_SHORT).show();
+
+        }
+    }
+
+    /**
+     * OnClick method to launch the update user dialog fragment.
+     */
+    @OnClick(R.id.user_info_fragment_about_me_button)
+    public void onUpdateInfoButtonClicked(){
+        UpdateUserDialogFragment dialogFragment = UpdateUserDialogFragment.newInstance(mainUser);
+        dialogFragment.setTargetFragment(this, 0);
```

Comment:
same here, worth mentioning why `setTargetFragment` is needed, and how it's used, i.e. what method in this class gets called by the dialog fragment when it's done